test/parallel/test-runner-mocking.js

```diff
@@ -318,6 +318,68 @@ test('spy functions can be bound', (t) => {
   assert.strictEqual(sum.mock.restore(), undefined);
   assert.strictEqual(sum.bind(0)(2, 11), 13);
 });
+test('spies on async class functions', async (t) => {
+  class Runner {
+    async someTask(msg) {
+      return Promise.resolve(msg);
+    }
+
+    async method(msg) {
+      await this.someTask(msg);
+      return msg;
+    }
+  }
+  const msg = 'ok';
+  const obj = new Runner();
+  assert.strictEqual(await obj.method(msg), msg);
```

Comment:
Can you also assert that the method to be mocked only exists on the prototype, and not on the instance itself.

Reply:
did you mean something like this?
```js
Runner.prototype.someTask.call(Runner.prototype, "hey")
```

I didn't get why not check the instance itself